src/dataconnect/fileUtils.spec.ts

```diff
@@ -127,81 +129,254 @@ describe("getPlatformFromFolder", () => {
+    const appFolderHasJava = "/my/app/folder/has-java";
+    const appFolderHasKotlin = "/my/app/folder/has-kotlin";
+    const appFolderHasBoth = "/my/app/folder/has-both";
+    const cases: {
+      desc: string;
+      appDir: string;
+      output: any;
+    }[] = [
+      {
+        desc: "basic",
+        appDir: appFolderBase,
+        output: {
+          outputDir: "../dataconnect-generated/kotlin",
+          package: "connectors.test_connector",
+        },
+      },
+      {
+        desc: "has java folder",
+        appDir: appFolderHasJava,
+        output: {
+          outputDir: "../has-java/app/src/main/java",
+          package: "connectors.test_connector",
+        },
+      },
+      {
+        desc: "has kotlin folder",
+        appDir: appFolderHasKotlin,
+        output: {
+          outputDir: "../has-kotlin/app/src/main/kotlin",
+          package: "connectors.test_connector",
+        },
+      },
+      {
+        desc: "prefer kotlin folder over java folder",
+        appDir: appFolderHasBoth,
+        output: {
+          outputDir: "../has-both/app/src/main/kotlin",
+          package: "connectors.test_connector",
+        },
+      },
+      {
+        desc: "below connector",
+        appDir: appFolderBelowConnector,
+        output: {
+          outputDir: "belowConnector/dataconnect-generated/kotlin",
+          package: "connectors.test_connector",
+        },
+      },
+      {
+        desc: "outside",
+        appDir: appFolderOutside,
+        output: {
+          outputDir: "../../outside/dataconnect-generated/kotlin",
+          package: "connectors.test_connector",
+        },
+      },
+    ];
+    for (const c of cases) {
+      it(c.desc, () => {
+        mockfs({
+          [appFolderHasJava + "/app/src/main/java"]: {},
+          [appFolderHasKotlin + "/app/src/main/kotlin"]: {},
+          [appFolderHasBoth + "/app/src/main/java"]: {},
+          [appFolderHasBoth + "/app/src/main/kotlin"]: {},
+        });
+        const modifiedYaml = generateSdkYaml(
+          Platform.ANDROID,
+          sampleConnectorYaml,
+          connectorYamlFolder,
+          c.appDir,
+        );
+        expect(modifiedYaml.generate?.kotlinSdk).to.deep.equal(c.output);
+      });
+    }
   });
 
-  it("should add Dart SDK generation for DART platform", () => {
-    const modifiedYaml = generateSdkYaml(
-      Platform.DART,
-      sampleConnectorYaml,
-      connectorYamlFolder,
-      appFolder,
-    );
-    expect(modifiedYaml.generate?.dartSdk).to.deep.equal({
-      outputDir: "../dataconnect-generated",
-      package: "test_connector",
-    });
+  describe("Dart platform should add Dart SDK Generation", () => {
+    const cases: {
+      desc: string;
+      appDir: string;
+      output: any;
+    }[] = [
+      {
+        desc: "basic",
+        appDir: appFolderBase,
+        output: {
+          outputDir: "../dataconnect-generated/dart/test-connector",
+          package: "test-connector",
```

Comment:
This should be an underscore for Dart, fwiw

Reply:
Done